composer/callbacks/checkpoint_saver.py

```diff
@@ -40,6 +48,54 @@
 _TORCH_DISTRIBUTED_CHECKPOINTS_METADATA_FILENAME = '.metadata'
 
 
+def _upload_symlink_file(
+    remote_backend_name: str,
+    backend_kwargs: dict[str, Any],
+    remote_symlink_file_name: str,
+    local_symlink_file_name: str,
+    num_attempts: int,
+    remote_checkpoint_file_names: list[str],
+    main_process_pid: int,
+    max_wait_time_in_seconds: int = 3600,
+    wait_before_next_try_in_seconds: float = 30,
+):
+    """Wait the checkpoint file uploading to finish and start symlink file uploading."""
+    start_time = time.time()
+    object_store = build_remote_backend(remote_backend_name, backend_kwargs)
+
+    for remote_file_name in remote_checkpoint_file_names:
+        while True:
+            # Return if parent process exits
+            try:
+                os.kill(main_process_pid, 0)
+            except OSError:
+                return
+            try:
+                object_store.get_object_size(remote_file_name)
+                break
```

Comment:
why this fn? can u not just check if it exists w/o checking size?

Reply:
do you have any API i can use in your mind? I dont' find a function that can check if a remote object exists [here](https://github.com/mosaicml/composer/blob/dev/composer/utils/object_store/object_store.py) . 

The `get_object_size` is how it's used to check if an object exists or not [here](https://github.com/mosaicml/composer/blob/25b656f4a97606fe1bf7edcd806bf5275311df0f/composer/utils/remote_uploader.py#L44)